urllib3/packages/ssl_match_hostname/__init__.py

```diff
@@ -1,5 +1,11 @@
+import sys
+
 try:
-    # Python 3.2+
+    # Our match_hostname function is the same as 3.5's, so we only want to
+    # import the match_hostname function if it's at least that good.
+    if sys.version_info < (3, 5):
```

Comment:
Not quite. We aren't carrying custom modifications to `match_hostname`, we're carrying basically the Python 3.5 `match_hostname` function with extra code to handle a missing `ipaddress` module. Basically, we're now behaving as though the Python 3.5 `match_hostname` function is the oldest version we will allow.